admin/create/city/city.controller.js

```diff
@@ -6,38 +6,68 @@
         .module('admin')
         .controller('CreateCityController', CreateCityController);
 
-    function CreateCityController ($log, $http, NotificationService, Endpoints) {
+    function CreateCityController($log, NotificationService, citiesProm, regionsProm, CreatePlacesService, $routeSegment) {
         const vm = this;
 
         vm.cities = [];
-        vm.newCity = '';
+        vm.regions = [];
+        vm.newCity = {};
 
         vm.createNewCity = createNewCity;
 
         activate();
-        
+
         ////
 
-        function activate () {
+        function activate() {
             $log.debug('Init CreateCityController ...');
             getCities();
+            getRegions();
         }
 
         function getCities() {
-            $http.get(Endpoints.CITIES).then((res) => {
-               
-                if(res.data.success) {
-                    vm.cities = res.data.data;
+            citiesProm.$promise.then((res) => {
+                if (res.success) {
+                    vm.cities = res.data;
                     $log.debug('response', vm.cities);
-                }else{
-                    NotificationService.error(res.data.data.message);
+                } else {
+                    NotificationService.error(res.message);
                 }
             });
         }
+        function getRegions() {
+            regionsProm.$promise.then((res) => {
+                if (res.success) {
+                    vm.regions = res.data;
+                    $log.debug('response', vm.regions);
+                } else {
+                    NotificationService.error(res.message);
+                }
+            });
+        }
+
+        function createNewCity() {
+            $log.debug('trying send data:', vm.newCity);
 
-        function createNewCity(e) {
-            e.preventDefault();
-            $log.debug('createCity', vm.newCity);
+            if (isNaN(vm.newCity.regionId) || !vm.newCity.name) {
+                $log.debug('wrong arguments:', vm.newCity);
+                return;
+            }
+
+            let params = {
+                "regionId": vm.newCity.regionId,
+                "name": vm.newCity.name
+            };
+
+            CreatePlacesService.newCity(params).$promise.then(res => {
```

Comment:
There is no "special" convention for naming services. But in general, we get the name of the service by concatenating "_Name of entity_" + "_Service_" word.
For example: **UserService**, **CustomerService**, **CompanyService**, becouse **User**, **Customer** and **Company** are entities. 

**CreatePlacesService** sounds like a method name, because CreatePlaces is not an entity. So you should rename it to **PlaceService**. It will be more correctly. Acordingly, the method will be called **PlaceService**.createCity(), .createCounty() etc.